hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/snapshot/SnapshotUtils.java

```diff
@@ -282,4 +320,47 @@ private static boolean isSameAsLatestOmKeyInfo(List<OmKeyInfo> omKeyInfos,
     }
     return false;
   }
+
+  public static SnapshotInfo getLatestGlobalSnapshotInfo(OzoneManager ozoneManager,
+                                                         SnapshotChainManager snapshotChainManager) throws IOException {
+    Optional<UUID> latestGlobalSnapshot  = Optional.ofNullable(snapshotChainManager.getLatestGlobalSnapshotId());
+    return latestGlobalSnapshot.isPresent() ? getSnapshotInfo(ozoneManager, snapshotChainManager,
+        latestGlobalSnapshot.get()) : null;
+  }
+
+  public static SnapshotInfo getLatestSnapshotInfo(String volumeName, String bucketName,
+                                                   OzoneManager ozoneManager,
+                                                   SnapshotChainManager snapshotChainManager) throws IOException {
+    Optional<UUID> latestPathSnapshot = Optional.ofNullable(
+        getLatestSnapshotId(volumeName, bucketName, snapshotChainManager));
+    return latestPathSnapshot.isPresent() ?
+        getSnapshotInfo(ozoneManager, snapshotChainManager, latestPathSnapshot.get()) : null;
+  }
+
+  public static UUID getLatestSnapshotId(String volumeName, String bucketName,
+                                         SnapshotChainManager snapshotChainManager) throws IOException {
+    String snapshotPath = volumeName + OM_KEY_PREFIX + bucketName;
+    return snapshotChainManager.getLatestPathSnapshotId(snapshotPath);
+  }
+
+  // Validates previous snapshotId given a snapshotInfo or volumeName & bucketName. Incase snapshotInfo is null, this
+  // would be considered as AOS and previous snapshot becomes the latest snapshot in the global snapshot chain.
+  // Would throw OMException if validation fails otherwise function would pass.
+  public static void validatePreviousSnapshotId(SnapshotInfo snapshotInfo,
+                                                SnapshotChainManager snapshotChainManager,
+                                                UUID expectedPreviousSnapshotId) throws IOException {
+    try {
+      UUID previousSnapshotId = snapshotInfo == null ? snapshotChainManager.getLatestGlobalSnapshotId() :
+          SnapshotUtils.getPreviousSnapshotId(snapshotInfo, snapshotChainManager);
+      if (!Objects.equals(expectedPreviousSnapshotId, previousSnapshotId)) {
+        throw new OMException("Snapshot validation failed. Expected previous snapshotId : " +
+                expectedPreviousSnapshotId + " but was " + previousSnapshotId,
+                OMException.ResultCodes.INVALID_REQUEST);
+      }
+    } catch (IOException e) {
+      LOG.error("Error while validating previous snapshot for snapshot: {}",
+          snapshotInfo == null ? null : snapshotInfo.getName(), e);
+      throw e;
+    }
+  }
```

Comment:
This catch is simply catching the exception thrown at like 356. I would suggest to remove this catch. Also, this is duplicate logging. You are logging the same [OMKeyPurgeRequest](https://github.com/apache/ozone/pull/7193/files#diff-f706c5f3b937e2ef0914b2f3e3d691552d8a651c3acf01c60baef6b3b0a1aff7R92) and [OMDirectoriesPurgeRequestWithFSO](https://github.com/apache/ozone/pull/7193/files#diff-9c21e51e11862689181392f4e84c7303e44450cbdb810f7a5dba737ce31f7c33R103). You can simply remove it. If you still want to log it, move it just before line 356 and without exception.

Reply:
Not exactly, this IOException would be thrown while getting latest snapshot from snapshot chain as well. Anyways removed the redundant error log